tools/database/src/analyze_delayed_receipt.rs

```diff
@@ -0,0 +1,123 @@
+use clap::Parser;
+use near_store::flat::FlatStorageManager;
+use near_store::{get_delayed_receipt_indices, ShardTries, StateSnapshotConfig, TrieConfig};
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::rc::Rc;
+
+use near_chain::ChainStore;
+use near_chain_configs::GenesisValidationMode;
+use nearcore::config::load_config;
+
+use near_primitives::hash::CryptoHash;
+use near_primitives::shard_layout::{ShardLayout, ShardUId};
+use near_primitives::types::BlockHeight;
+
+use crate::block_iterators::{
+    make_block_iterator_from_command_args, CommandArgs, LastNBlocksIterator,
+};
+use nearcore::open_storage;
+
+#[derive(Parser)]
+pub(crate) struct AnalyzeDelayedReceiptCommand {
+    /// Analyse the last N blocks in the blockchain
+    #[arg(long)]
+    last_blocks: Option<u64>,
+
+    /// Analyse blocks from the given block height, inclusive
+    #[arg(long)]
+    from_block_height: Option<BlockHeight>,
+
+    /// Analyse blocks up to the given block height, inclusive
+    #[arg(long)]
+    to_block_height: Option<BlockHeight>,
+}
+
+impl AnalyzeDelayedReceiptCommand {
+    pub(crate) fn run(&self, home: &PathBuf) -> anyhow::Result<()> {
+        let mut near_config = load_config(home, GenesisValidationMode::Full).unwrap();
+        let node_storage = open_storage(&home, &mut near_config).unwrap();
+        let store = node_storage.get_split_store().unwrap_or_else(|| node_storage.get_hot_store());
+        let chain_store = Rc::new(ChainStore::new(
+            store.clone(),
+            near_config.genesis.config.genesis_height,
+            false,
+        ));
+        let shard_layout = ShardLayout::get_simple_nightshade_layout();
+        // hard code the shard ids
+        let shard_ids = vec![0, 1, 2, 3]; 
```

Comment:
nit: would be nicer to get the shard layout and shard ids for each block individually then you always get the right one and you also don't need to handle the shard layout changing

Reply:
I think with epoch manager you can do something like the following:
block -> epoch id -> shard layout